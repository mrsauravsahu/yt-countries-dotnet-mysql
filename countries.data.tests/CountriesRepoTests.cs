using System;
using System.Threading.Tasks;
using countries.data.repos;
using Microsoft.EntityFrameworkCore;
using Snapshooter.Xunit;
using Xunit;

namespace countries.data.tests
{
    public class CountriesRepoTests
    {
        private readonly CountriesRepo countriesRepo;

        public CountriesRepoTests()
        {
            var optionsBuilder = new DbContextOptionsBuilder<CountriesContext>();
            optionsBuilder.UseMySql("host=sample", new MySqlServerVersion("8.0"));

            var countriesContext = new CountriesContext(optionsBuilder.Options);


            this.countriesRepo = new CountriesRepo(countriesContext);
        }

        [Fact]
        public async Task GetCountriesWithContinentAsync()
        {
            var countries = await countriesRepo.GetCountriesWithContinentAsync();
            Snapshot.Match(countries.ToQueryString());
        }
    }
}
