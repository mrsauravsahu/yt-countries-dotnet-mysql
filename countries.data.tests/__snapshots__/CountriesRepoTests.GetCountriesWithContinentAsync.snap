SELECT `c`.`Id`, `c`.`ContinentId`, `c`.`CreatedAt`, `c`.`Name`, `c0`.`Id`, `c0`.`CreatedAt`, `c0`.`Name`
FROM `Countries` AS `c`
LEFT JOIN `Continents` AS `c0` ON `c`.`ContinentId` = `c0`.`Id`
ORDER BY `c`.`Name`
